gef.py

```diff
@@ -10482,22 +10482,83 @@ def lookup_command(self, cmd):
 
         return None
 
+@register_command
+class AliasesCommand(GenericCommand):
+    """Base command to add, remove, or list aliases"""
 
-class GefAliases(gdb.Command):
-    """List all custom aliases."""
+    _cmdline_ = "aliases"
+    _syntax_  = "{:s} (add|rm|list)".format(_cmdline_)
 
     def __init__(self):
-        super().__init__("aliases", gdb.COMMAND_OBSCURE, gdb.COMPLETE_NONE)
+        super().__init__(prefix=True)
         return
 
-    def invoke(self, args, from_tty):
-        self.dont_repeat()
+    def do_invoke(self, argv):
+        self.usage()
+        return
+
+@register_command
+class AliasesAddCommand(AliasesCommand):
+    """Command to add aliases"""
+
+    _cmdline_ = "aliases add"
+    _syntax_  = "{0} [ALIAS] [COMMAND]".format(_cmdline_)
+    _example_ = "{0} scope telescope".format(_cmdline_)
+
+    def __init__(self):
+        super().__init__()
+        return
+
+    def do_invoke(self, argv):
+        if (len(argv) < 2):
+            self.usage()
+            return
+        GefAlias(argv[0], " ".join(argv[1:]))
+        GefSaveCommand().invoke(None, False)
+        return
+
+@register_command
+class AliasesRmCommand(AliasesCommand):
+    """Command to remove aliases"""
+
+    _cmdline_ = "aliases rm"
+    _syntax_ = "{0} [ALIAS]".format(_cmdline_)
+
+    def __init__(self):
+        super().__init__()
+        return
+
+    def do_invoke(self, argv):
+        global __aliases__
+        if (len(argv) != 1):
+            self.rm_usage()
+            return
+        numaliases = len(__aliases__)
+        __aliases__ = [alias for alias in __aliases__ if alias._alias != argv[0]]
+        if numaliases == len(__aliases__):
+             err("{0} not found in aliases.".format(argv[0]))
+             return
+        GefSaveCommand().invoke(None, False)
+        # TODO: reload GEF so that the alias no longer works in the current session?
```

Comment:
this probably won't work. Could print a message like "you must restart GDB for alias removals to apply" or something.